tests/test_meta_tensor.py

```diff
@@ -0,0 +1,232 @@
+        m = m.to("cuda")
+        self.check(m, orig, ids=False, device="cuda")
+        m = m.cpu()
+        self.check(m, orig, ids=False, device="cpu")
+        m = m.cuda()
+        self.check(m, orig, ids=False, device="cuda")
+        m = m.to("cpu")
+        self.check(m, orig, ids=False, device="cpu")
+        m = m.to(device="cuda")
+        self.check(m, orig, ids=False, device="cuda")
+        m = m.to(device="cpu")
+        self.check(m, orig, ids=False, device="cpu")
+
+    @parameterized.expand(TESTS)
+    def test_copy(self, device, dtype):
+        m, _ = self.get_im(device=device, dtype=dtype)
+        # shallow copy
+        a = m
+        self.check(a, m, ids=True)
+        # deepcopy
+        a = deepcopy(m)
+        self.check(a, m, ids=False)
+        # clone
+        a = m.clone()
+        self.check(a, m, ids=False)
+
+    @parameterized.expand(TESTS)
+    def test_add(self, device, dtype):
+        m1, t1 = self.get_im(device=device, dtype=dtype)
+        m2, t2 = self.get_im(device=device, dtype=dtype)
+        self.check(m1 + m2, t1 + t2, ids=False)
+        self.check(torch.add(m1, m2), t1 + t2, ids=False)
+        self.check(torch.add(input=m1, other=m2), t1 + t2, ids=False)
+        self.check(torch.add(m1, other=m2), t1 + t2, ids=False)
+        m3 = deepcopy(m2)
+        t3 = deepcopy(m2)
+        m3 += 3
+        t3 += 3
+        self.check(m3, t3, ids=False)
+        # check torch.Tensor+MetaTensor and MetaTensor+torch.Tensor
+        self.check(torch.add(m1, t2), t1 + t2, ids=False)
+        self.check(torch.add(t2, m1), t1 + t2, ids=False)
+
+    @parameterized.expand(TEST_DEVICES)
+    def test_conv(self, device):
+        im, _ = self.get_im((1, 3, 10, 8, 12), device=device)
+        conv = torch.nn.Conv3d(im.shape[1], 5, 3)
+        conv.to(device)
+        out = conv(im)
+        self.check(out, im, shape=False, vals=False, ids=False)
+
+    @parameterized.expand(TESTS)
+    def test_stack(self, device, dtype):
+        numel = 3
+        ims = [self.get_im(device=device, dtype=dtype)[0] for _ in range(numel)]
+        stacked = torch.stack(ims)
+        self.assertIsInstance(stacked, MetaTensor)
+        assert_allclose(stacked.affine, ims[0].affine)
+        self.assertEqual(stacked.meta, ims[0].meta)
+
+    def test_get_set_meta_fns(self):
+        set_track_meta(False)
+        self.assertEqual(get_track_meta(), False)
+        set_track_meta(True)
+        self.assertEqual(get_track_meta(), True)
+        set_track_transforms(False)
+        self.assertEqual(get_track_transforms(), False)
+        set_track_transforms(True)
+        self.assertEqual(get_track_transforms(), True)
+
+    @parameterized.expand(TEST_DEVICES)
+    def test_torchscript(self, device):
+        shape = (1, 3, 10, 8)
+        im, _ = self.get_im(shape, device=device)
+        conv = torch.nn.Conv2d(im.shape[1], 5, 3)
+        conv.to(device)
+        im_conv = conv(im)
+        traced_fn = torch.jit.trace(conv, im)
```

Comment:
perhaps we only support this?
```suggestion
        traced_fn = torch.jit.trace(conv, im.as_tensor())
```